controllers/recipes.js

```diff
@@ -0,0 +1,170 @@
+  },
+
+  createReview: async (request, response) => {
+    const recipe = await Recipe.findById(request.params.id);
+    const user = request.user;
+
+    const {timesMade, rating, picturePath} = request.body;
+
+    const review = new Review({
+      recipeId: recipe._id,
+      reviewer: user._id,
+      timesMade,
+      rating,
+      picturePath,
+    });
+    const createdReview = await review.save();
+    user.reviews = user.reviews.concat(createdReview._id);
+    await user.save();
+    recipe.review = recipe.review.concat(createdReview._id);
+    await recipe.save();
+
+    response.status(201).json(createdReview);
+  },
+  createComment: async (request, response) => {
+    const recipe = await Recipe.findById(request.params.id);
+    const user = request.user;
+    const {parentId, text} = request.body;
+
+    const comment = new Comments({
+      recipeId: recipe._id,
+      userId: user._id,
+      parentId,
+      text,
+    });
+
+    const createdComment = await comment.save();
+
+    recipe.comments = recipe.comments.concat(createdComment._id);
+    await recipe.save();
+    response.status(201).json(createdComment);
+  },
+
+  /* READ */
+
+  getFeedRecipes: async (request, response) => {
+    try {
+      const recipe = await Recipe.find();
+      response.status(200).json(recipe);
+    } catch (err) {
+      response.status(404).json({message: err.message});
+    }
+  },
+  getRecipe: async (request, response) => {
+    try {
+      const recipe = await Recipe.findById(request.params.id)
+        .populate("review", {
+          reviewer: 1,
+          timesMade: 1,
+          rating: 1,
+          picturePath: 1,
+        })
+        .populate("comments", {userId: 1, parentId: 1, text: 1});
+
+      response.status(200).json(recipe);
+    } catch (err) {
+      response.status(404).json({message: err.message});
+    }
+  },
+
+  getUserRecipes: async (request, response) => {
+    try {
+      const user = await User.findById(request.params.id).populate("recipe", {
+        title: 1,
+        picturePath: 1,
+        tags: 1,
+        userId: 1,
+      });
+      response.status(200).json(user);
+    } catch (err) {
+      console.log(err);
+    }
+  },
+
+  /* UPDATE */
+
+  addToSaveRecipe: async (request, response) => {
+    try {
+      const {id} = request.params;
+
+      const user = await User.findById(request.user._id);
+      const isLiked = user.savedRecipe.get(id);
+
+      if (isLiked) {
+        user.savedRecipe.delete(id);
+      } else {
+        user.savedRecipe.set(id, true);
+      }
+
+      const updatedUser = await User.findByIdAndUpdate(
+        id,
+        {savedRecipe: user.savedRecipe},
+        {new: true}
+      );
+      response.status(200).json(updatedUser);
+    } catch (err) {
+      res.status(404).json({message: err.message});
+    }
+  },
+
+  /* DELETE */
+  deleteRecipe: async (request, response) => {
+    const recipe = await Recipe.findById(request.params.id);
```

Comment:
I don't think you deleted the recipe from the database

Reply:
It is normal not to delete? I'm not sure what is better :/

Reply:
I heard big companies don't actually delete LOL

Reply:
wait is it not right? I removed it from the user by declaring the recipe property and then the recipe object is removed?